File: Python3/1493. Longest Subarray of 1's After Deleting One Element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        l, ans, zero = 0, 0, -1
        for r, n in enumerate(nums):
            if n == 0:
                l = zero + 1
                zero = r
            ans = max(ans, r - l)

        return ans

File: Java/1493. Longest Subarray of 1's After Deleting One Element.java
class Solution {
    public int longestSubarray(int[] nums) {
        int ans = 0, l = 0, zero = -1;
        for (int r = 0; r < nums.length; r++) {
            if (nums[r] == 0) {
                l = zero + 1;
                zero = r;
            }
            ans = Math.max(ans, r - l);
        }
        return ans;
    }
}